### PR TITLE
Make User entity extendable

### PR DESCRIPTION
**Pull request by matthijsch**

### CODE REVIEW

File: src/DependencyInjection/Compiler/ResolveTargetEntityPass.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace ConnectHolland\UserBundle\DependencyInjection\Compiler;
+
+use ConnectHolland\UserBundle\Entity\UserInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @codeCoverageIgnore WIP
```
Comment (reyostallenberg) [inline src/DependencyInjection/Compiler/ResolveTargetEntityPass.php]: Should have an explanation of why it's ignored, or it should be removed and the tests should cover it.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -37,6 +37,7 @@ public function getConfigTreeBuilder()
                     ->prototype('scalar')->end()
                     ->defaultValue(['main'])
                 ->end()
+                ->scalarNode('user_class')->cannotBeEmpty()->end()
```
Comment (reyostallenberg) [inline src/DependencyInjection/Configuration.php]: Default to an/the entity in this bundle?

File: src/DependencyInjection/ConnecthollandUserExtension.php [hunk 3]
```diff
@@ -27,6 +27,10 @@ class ConnecthollandUserExtension extends Extension implements ExtensionInterfac
 {
     public function load(array $configs, ContainerBuilder $container)
     {
+        $configuration = new Configuration();
+        $config        = $this->processConfiguration($configuration, $configs);
+        $container->setParameter('connectholland_user.user_class', $config['user_class']);
```
Comment (reyostallenberg) [inline src/DependencyInjection/ConnecthollandUserExtension.php]: This should not be needed, you should 'fix' the service definition of the service that needs it. IIRC replaceArgument call on it's definition

File: src/Entity/User.php [hunk 4]
```diff
@@ -25,43 +25,43 @@ class User implements UserInterface
      * @ORM\GeneratedValue()
      * @ORM\Column(type="integer")
      */
-    private $id;
```
Comment (reyostallenberg) [inline src/Entity/User.php]: Do these all need to be widened?

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (matthijsch) [reply src/DependencyInjection/Configuration.php]: Can't do this as the user class is a superclass which is not an entity of itself.

**Comment by reyostallenberg** [comment]
So now you have to create an entity in your app always. The idea behind MappedSupperClass is extending, which I don't like. Can't you just implement the interface? And maybe extract some stuff to a trait for convenience.
Can you ask other people's opinion?

**Comment by matthijsch** [comment]
Ideally it shouldn't be required to create an entity in your app, I agree with that, but it should be possible to extend the user entity, because how could you otherwise create a relation with for example a Profile entity. I can't figure out a way to make it possible to extend without using MappedSuperClass.

**Pull request merged**